### PR TITLE
Add test case for handling empty response from an endpoint by callout mediator

### PR DESCRIPTION
**Pull request by ThishaniLucas**
PR sent to ESB : https://github.com/wso2/product-esb/pull/460

### CODE REVIEW

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/callOut/ESBJAVA4452EmptySuccessResponseTestCase.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import org.testng.annotations.Test;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+import org.wso2.esb.integration.common.utils.clients.SimpleHttpClient;
+
+import static org.testng.Assert.assertEquals;
+
+/**
+ * Test case to check whether callout mediator is handling an empty reponse from the endpoint
+ */
+public class ESBJAVA4452EmptySuccessResponseTestCase extends ESBIntegrationTest {
```
Comment (sdkottegoda) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/callOut/ESBJAVA4452EmptySuccessResponseTestCase.java]: Let's not have the name as 'ESBJAVA...' in the public branch. Shall we correct in all places?

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/callOut/ESBJAVA4452EmptySuccessResponseTestCase.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @Test(groups = "wso2.esb")
+    public void testForwardingWithInMemoryStore() throws Exception {
+        String requestPayload = "{\"hello\":\"world\"}";
+        SimpleHttpClient httpClient = new SimpleHttpClient();
+        String url = getApiInvocationURL("callouttest");
+        HttpResponse httpResponse = httpClient.doPost(url, null, requestPayload, "application/json");
+        String responsePayload = httpClient.getResponsePayload(httpResponse);
+        assertEquals(responsePayload, requestPayload, "matching request and response");
```
Comment (sdkottegoda) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/callOut/ESBJAVA4452EmptySuccessResponseTestCase.java]: The assert message should be modified

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/callOut/ESBJAVA4452EmptySuccessResponseTestCase.java (hunk 1 above)
Comment (ThishaniLucas) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/callOut/ESBJAVA4452EmptySuccessResponseTestCase.java]: done

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/callOut/ESBJAVA4452EmptySuccessResponseTestCase.java (hunk 2 above)
Comment (ThishaniLucas) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/callOut/ESBJAVA4452EmptySuccessResponseTestCase.java]: fixed
